src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -794,6 +818,18 @@ public static string Escape(string argument)
             return escaped;
         }
 
+        public static string EscapeQuoted(string argument)
```

Comment:
👍 on regex safety.

I'd like to move away from the current approach of passing everything in as args, I've already seen customers hit issues where they go beyond the args length limit for Process.

We should instead write out a targets file containing everything and call msbuild on that. That doesn't need to be part of this PR, I can make that change and it would eliminate the need for this escaping.